src/Compilers/Core/Portable/Operations/Operation.cs

```diff
@@ -36,6 +37,22 @@ public Operation(OperationKind kind, SyntaxNode syntax, ITypeSymbol type, Option
         public ITypeSymbol Type { get; }
 
         /// <summary>
+        /// The source language of the IOperation. Possible values are <see cref="LanguageNames.CSharp"/> and <see cref="LanguageNames.VisualBasic"/>.
+        /// </summary>
+
+        public string Language
+        {
+            get
+            {
+                // TODO: it is an eventual goal to support analyzing IL. At that point, we'll need to detect a null
+                // syntax and add a new field to LanguageNames for IL. Until then, though, we'll just assume that
+                // syntax is not null and return its language.
+                Debug.Assert(Syntax != null);
```

Comment:
assert provides no value.  we're just firing an assert one line earlier in debug.  but in release we're still going to crash.  i would just remove and keep the comment.